From 2313eb7158479646ec9f85893453951c3b4b58df Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=81kos=20Uzonyi?= <uzonyi.akos@gmail.com>
Date: Fri, 19 Jun 2020 12:06:43 +0200
Subject: [PATCH 130/138] Modify %process class: trace syscalls associated with
 process lifecycle

The description is rephrased as:
"Trace system calls associated with process lifecycle
(creation, exec, termination)."

kill, tkill, tgkill, pidfd_send_signal and rt_sigqueueinfo are added,
arch_prctl and unshare are removed from the %process class.

syscallent*.h files are updated automatically by:
find linux -name syscallent*.h -exec \
sed -i -r -e '
    /(arch_prctl|unshare)/ {
        s/(\{[^,]*,\t[^,]*)TP\|/\1/
        s/(\{[^,]*,\t[^,]*)\|TP/\1/
        s/(\{[^,]*,\t[^,]*)TP,/\10,/
    }
' -e '
    /(kill|pidfd_send_signal|rt_sigqueueinfo)/ {
        s/(\{[^,]*,\t[^0][^,]*)/\1|TP/
        s/(\{[^,]*,\s*)0/\1TP/
    }
' {} +

[ldv: document this change in NEWS.]

* strace.1.in: Rephrase %process description.
* linux/32/syscallent.h: Add TP flag to *kill, pidfd_send_signal and
rt_sigqueueinfo; remove TP flag from arch_prctl and unshare.
* linux/64/syscallent.h: Likewise.
* linux/alpha/syscallent.h: Likewise.
* linux/arm/syscallent.h: Likewise.
* linux/avr32/syscallent.h: Likewise.
* linux/bfin/syscallent.h: Likewise.
* linux/hppa/syscallent.h: Likewise.
* linux/i386/syscallent.h: Likewise.
* linux/ia64/syscallent.h: Likewise.
* linux/m68k/syscallent.h: Likewise.
* linux/microblaze/syscallent.h: Likewise.
* linux/mips/syscallent-compat.h: Likewise.
* linux/mips/syscallent-n32.h: Likewise.
* linux/mips/syscallent-n64.h: Likewise.
* linux/mips/syscallent-o32.h: Likewise.
* linux/powerpc/syscallent.h: Likewise.
* linux/powerpc64/syscallent.h: Likewise.
* linux/s390/syscallent.h: Likewise.
* linux/s390x/syscallent.h: Likewise.
* linux/sh/syscallent.h: Likewise.
* linux/sh64/syscallent.h: Likewise.
* linux/sparc/syscallent.h: Likewise.
* linux/sparc64/syscallent.h: Likewise.
* linux/syscallent-common.h: Likewise.
* linux/x32/syscallent.h: Likewise.
* linux/x86_64/syscallent.h: Likewise.
* linux/xtensa/syscallent.h: Likewise.
* NEWS: Mention this change.

Conflicts:
	NEWS
	linux/i386/syscallent.h
---
 linux/32/syscallent.h          | 10 +++++-----
 linux/64/syscallent.h          | 10 +++++-----
 linux/alpha/syscallent.h       | 12 ++++++------
 linux/arm/syscallent.h         | 10 +++++-----
 linux/avr32/syscallent.h       | 10 +++++-----
 linux/bfin/syscallent.h        | 10 +++++-----
 linux/hppa/syscallent.h        | 10 +++++-----
 linux/i386/syscallent.h        | 12 ++++++------
 linux/ia64/syscallent.h        | 10 +++++-----
 linux/m68k/syscallent.h        | 10 +++++-----
 linux/microblaze/syscallent.h  | 10 +++++-----
 linux/mips/syscallent-compat.h | 10 +++++-----
 linux/mips/syscallent-n32.h    | 10 +++++-----
 linux/mips/syscallent-n64.h    | 10 +++++-----
 linux/mips/syscallent-o32.h    | 10 +++++-----
 linux/powerpc/syscallent.h     | 10 +++++-----
 linux/powerpc64/syscallent.h   | 10 +++++-----
 linux/s390/syscallent.h        | 10 +++++-----
 linux/s390x/syscallent.h       | 10 +++++-----
 linux/sh/syscallent.h          | 10 +++++-----
 linux/sh64/syscallent.h        | 10 +++++-----
 linux/sparc/syscallent.h       | 10 +++++-----
 linux/sparc64/syscallent.h     | 10 +++++-----
 linux/syscallent-common.h      |  2 +-
 linux/x32/syscallent.h         | 14 +++++++-------
 linux/x86_64/syscallent.h      | 12 ++++++------
 linux/xtensa/syscallent.h      | 10 +++++-----
 strace.1.in                    |  4 ++--
 28 files changed, 138 insertions(+), 138 deletions(-)

diff --git a/linux/32/syscallent.h b/linux/32/syscallent.h
index 5e1f06d..c74ab18 100644
--- a/linux/32/syscallent.h
+++ b/linux/32/syscallent.h
@@ -109,7 +109,7 @@
 [ 94] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [ 95] = { 5,	TP,		SEN(waitid),			"waitid"		},
 [ 96] = { 1,	0,		SEN(set_tid_address),		"set_tid_address"	},
-[ 97] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[ 97] = { 1,	0,		SEN(unshare),			"unshare"		},
 /* [ 98] futex */
 [ 99] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [100] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
@@ -141,16 +141,16 @@
 [126] = { 1,	0,		SEN(sched_get_priority_min),	"sched_get_priority_min"},
 /* [127] sched_rr_get_interval */
 [128] = { 0,	0,		SEN(restart_syscall),		"restart_syscall"	},
-[129] = { 2,	TS,		SEN(kill),			"kill"			},
-[130] = { 2,	TS,		SEN(kill),			"tkill"			},
-[131] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[129] = { 2,	TS|TP,		SEN(kill),			"kill"			},
+[130] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
+[131] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [132] = { 2,	TS,		SEN(sigaltstack),		"sigaltstack"		},
 [133] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [134] = { 4,	TS,		SEN(rt_sigaction),		"rt_sigaction"		},
 [135] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [136] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 /* [137] rt_sigtimedwait */
-[138] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[138] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [139] = { 0,	TS,		SEN(rt_sigreturn),		"rt_sigreturn"		},
 [140] = { 3,	0,		SEN(setpriority),		"setpriority"		},
 [141] = { 2,	0,		SEN(getpriority),		"getpriority"		},
diff --git a/linux/64/syscallent.h b/linux/64/syscallent.h
index 9531a05..3fb1305 100644
--- a/linux/64/syscallent.h
+++ b/linux/64/syscallent.h
@@ -102,7 +102,7 @@
 [ 94] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [ 95] = { 5,	TP,		SEN(waitid),			"waitid"		},
 [ 96] = { 1,	0,		SEN(set_tid_address),		"set_tid_address"	},
-[ 97] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[ 97] = { 1,	0,		SEN(unshare),			"unshare"		},
 [ 98] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [ 99] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [100] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
@@ -134,16 +134,16 @@
 [126] = { 1,	0,		SEN(sched_get_priority_min),	"sched_get_priority_min"},
 [127] = { 2,	0,		SEN(sched_rr_get_interval_time64),"sched_rr_get_interval"},
 [128] = { 0,	0,		SEN(restart_syscall),		"restart_syscall"	},
-[129] = { 2,	TS,		SEN(kill),			"kill"			},
-[130] = { 2,	TS,		SEN(kill),			"tkill"			},
-[131] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[129] = { 2,	TS|TP,		SEN(kill),			"kill"			},
+[130] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
+[131] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [132] = { 2,	TS,		SEN(sigaltstack),		"sigaltstack"		},
 [133] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [134] = { 4,	TS,		SEN(rt_sigaction),		"rt_sigaction"		},
 [135] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [136] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [137] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[138] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[138] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [139] = { 0,	TS,		SEN(rt_sigreturn),		"rt_sigreturn"		},
 [140] = { 3,	0,		SEN(setpriority),		"setpriority"		},
 [141] = { 2,	0,		SEN(getpriority),		"getpriority"		},
diff --git a/linux/alpha/syscallent.h b/linux/alpha/syscallent.h
index b0d680b..93f0b0e 100644
--- a/linux/alpha/syscallent.h
+++ b/linux/alpha/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 5,	0,		SEN(printargs),			"osf_chflags"		}, /* not implemented */
 [ 35] = { 5,	0,		SEN(printargs),			"osf_fchflags"		}, /* not implemented */
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 5,	TF|TST|TSTA,	SEN(printargs),			"osf_old_stat"		}, /* not implemented */
 [ 39] = { 2,	0,		SEN(setpgid),			"setpgid"		},
 [ 40] = { 5,	TF|TLST|TSTA,	SEN(printargs),			"osf_old_lstat"		}, /* not implemented */
@@ -153,7 +153,7 @@
 [143] = { 5,	0,		SEN(printargs),			"osf_sethostid"		}, /* not implemented */
 [144] = { 2,	0,		SEN(getrlimit),			"getrlimit"		},
 [145] = { 2,	0,		SEN(setrlimit),			"setrlimit"		},
-[146] = { 5,	0,		SEN(printargs),			"osf_old_killpg"	}, /* not implemented */
+[146] = { 5,	TP,		SEN(printargs),			"osf_old_killpg"	}, /* not implemented */
 [147] = { 0,	0,		SEN(setsid),			"setsid"		},
 [148] = { 4,	TF,		SEN(quotactl),			"quotactl"		},
 [149] = { 5,	0,		SEN(printargs),			"osf_oldquota"		}, /* not implemented */
@@ -300,7 +300,7 @@
 [353] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [354] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [355] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[356] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[356] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [357] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [358] = { 5,	TD,		SEN(select),			"select"		},
 [359] = { 2,	TCL,		SEN(gettimeofday),		"gettimeofday"		},
@@ -325,7 +325,7 @@
 [378] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
 [379] = { 3,	TD,		SEN(readahead),			"readahead"		},
 [380] = { },
-[381] = { 2,	TS,		SEN(kill),			"tkill"			},
+[381] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [382] = { 5,	TF,		SEN(setxattr),			"setxattr"		},
 [383] = { 5,	TF,		SEN(setxattr),			"lsetxattr"		},
 [384] = { 5,	TD,		SEN(fsetxattr),			"fsetxattr"		},
@@ -367,7 +367,7 @@
 [421] = { 2,	TCL,		SEN(clock_getres_time64),	"clock_getres"		},
 [422] = { 4,	0,		SEN(clock_nanosleep_time64),	"clock_nanosleep"	},
 [423] = { 4,	TI,		SEN(semtimedop_time64),		"semtimedop"		},
-[424] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[424] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [425] = { 2,	TF|TST|TSTA,	SEN(stat64),			"stat64"		},
 [426] = { 2,	TF|TLST|TSTA,	SEN(lstat64),			"lstat64"		},
 [427] = { 2,	TD|TFST|TSTA,	SEN(fstat64),			"fstat64"		},
@@ -408,7 +408,7 @@
 [462] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [463] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [464] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[465] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[465] = { 1,	0,		SEN(unshare),			"unshare"		},
 [466] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [467] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [468] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/arm/syscallent.h b/linux/arm/syscallent.h
index e9eee9f..73497b6 100644
--- a/linux/arm/syscallent.h
+++ b/linux/arm/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 6,	TD,		SEN(pread),			"pread64"		},
 [181] = { 6,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -245,7 +245,7 @@
 [235] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [236] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [237] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[238] = { 2,	TS,		SEN(kill),			"tkill"			},
+[238] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [239] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [240] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [241] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -275,7 +275,7 @@
 [265] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [266] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [267] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[268] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[268] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [269] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [270] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [271] = { 3,	0,		SEN(printargs),			"pciconfig_iobase"	},
@@ -344,7 +344,7 @@
 [334] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [335] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [336] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[337] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[337] = { 1,	0,		SEN(unshare),			"unshare"		},
 [338] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [339] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [340] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/avr32/syscallent.h b/linux/avr32/syscallent.h
index 0b24353..b3cf2da 100644
--- a/linux/avr32/syscallent.h
+++ b/linux/avr32/syscallent.h
@@ -43,7 +43,7 @@
 [ 34] = { 1,	TF,		SEN(chroot),			"chroot"		},
 [ 35] = { 0,	0,		SEN(sync),			"sync"			},
 [ 36] = { 1,	TD,		SEN(fsync),			"fsync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -78,7 +78,7 @@
 [ 69] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [ 70] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [ 71] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[ 72] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[ 72] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [ 73] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [ 74] = { 2,	0,		SEN(sethostname),		"sethostname"		},
 [ 75] = { 2,	0,		SEN(setrlimit),			"setrlimit"		},
@@ -196,7 +196,7 @@
 [187] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [188] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [189] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[190] = { 2,	TS,		SEN(kill),			"tkill"			},
+[190] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [191] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [192] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [193] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -227,7 +227,7 @@
 [218] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [219] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [220] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[221] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[221] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [222] = { },
 [223] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [224] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
@@ -264,7 +264,7 @@
 [255] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [256] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [257] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[258] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[258] = { 1,	0,		SEN(unshare),			"unshare"		},
 [259] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [260] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [261] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/bfin/syscallent.h b/linux/bfin/syscallent.h
index 4deafe0..b93a844 100644
--- a/linux/bfin/syscallent.h
+++ b/linux/bfin/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 5,	TD,		SEN(pread),			"pread"			},
 [181] = { 5,	TD,		SEN(pwrite),			"pwrite"		},
@@ -244,7 +244,7 @@
 [235] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [236] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [237] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[238] = { 2,	TS,		SEN(kill),			"tkill"			},
+[238] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [239] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [240] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [241] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -277,7 +277,7 @@
 [268] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [269] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [270] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[271] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[271] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [272] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [273] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [274] = { 5,	0,		SEN(vserver),			"vserver"		},
@@ -316,7 +316,7 @@
 [307] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [308] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [309] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[310] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[310] = { 1,	0,		SEN(unshare),			"unshare"		},
 [311] = { 2,	0,		SEN(sram_alloc),		"sram_alloc"		},
 [312] = { 1,	0,		SEN(printargs),			"sram_free"		},
 [313] = { 3,	0,		SEN(printargs),			"dma_memcpy"		},
diff --git a/linux/hppa/syscallent.h b/linux/hppa/syscallent.h
index dca6c41..31341d4 100644
--- a/linux/hppa/syscallent.h
+++ b/linux/hppa/syscallent.h
@@ -40,7 +40,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 3,	TN,		SEN(accept),			"accept"		},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -181,7 +181,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 3,	TF,		SEN(chown),			"chown"			},
 [181] = { 5,	TN,		SEN(setsockopt),		"setsockopt"		},
@@ -211,7 +211,7 @@
 [205] = { 5,	0,		SEN(printargs),			"acl_set"		},
 [206] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
 [207] = { 4,	TD,		SEN(readahead),			"readahead"		},
-[208] = { 2,	TS,		SEN(kill),			"tkill"			},
+[208] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [209] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [210] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [211] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -262,7 +262,7 @@
 [256] = { 2,	TCL,		SEN(clock_gettime32),		"clock_gettime"		},
 [257] = { 2,	TCL,		SEN(clock_getres_time32),	"clock_getres"		},
 [258] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
-[259] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[259] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [260] = { 6,	TM,		SEN(mbind),			"mbind"			},
 [261] = { 5,	TM,		SEN(get_mempolicy),		"get_mempolicy"		},
 [262] = { 3,	TM,		SEN(set_mempolicy),		"set_mempolicy"		},
@@ -291,7 +291,7 @@
 [285] = { 4,	TD|TF,		SEN(readlinkat),		"readlinkat"		},
 [286] = { 3,	TD|TF,		SEN(fchmodat),			"fchmodat"		},
 [287] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
-[288] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[288] = { 1,	0,		SEN(unshare),			"unshare"		},
 [289] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [290] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [291] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/i386/syscallent.h b/linux/i386/syscallent.h
index 0427870..efe0ff7 100644
--- a/linux/i386/syscallent.h
+++ b/linux/i386/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 5,	TD,		SEN(pread),			"pread64"		},
 [181] = { 5,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -244,7 +244,7 @@
 [235] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [236] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [237] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[238] = { 2,	TS,		SEN(kill),			"tkill"			},
+[238] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [239] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [240] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [241] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -276,7 +276,7 @@
 [267] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [268] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [269] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[270] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[270] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [271] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [272] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [273] = { 5,	0,		SEN(vserver),			"vserver"		},
@@ -316,7 +316,7 @@
 [307] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [308] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [309] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[310] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[310] = { 1,	0,		SEN(unshare),			"unshare"		},
 [311] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [312] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [313] = { 6,	TD,		SEN(splice),			"splice"		},
@@ -390,7 +390,7 @@
 [381] = { 2,	0,		SEN(pkey_alloc),		"pkey_alloc"		},
 [382] = { 1,	0,		SEN(pkey_free),			"pkey_free"		},
 [383] = { 5,	TD|TF|TSTA,	SEN(statx),			"statx"			},
-[384] = { 2,	TP,		SEN(arch_prctl),		"arch_prctl"		},
+[384] = { 2,	0,		SEN(arch_prctl),		"arch_prctl"		},
 [385] = { 6,	0,		SEN(io_pgetevents_time32),	"io_pgetevents"		},
 [386] = { 4,	0,		SEN(rseq),			"rseq"			},
 /* room for arch specific calls */
diff --git a/linux/ia64/syscallent.h b/linux/ia64/syscallent.h
index 9098488..8aeda41 100644
--- a/linux/ia64/syscallent.h
+++ b/linux/ia64/syscallent.h
@@ -49,7 +49,7 @@
 [BASE_NR +  26] = { 0,	0,		SEN(sync),			"sync"			},
 [BASE_NR +  27] = { 1,	TD,		SEN(fsync),			"fsync"			},
 [BASE_NR +  28] = { 1,	TD,		SEN(fdatasync),			"fdatasync"		},
-[BASE_NR +  29] = { 2,	TS,		SEN(kill),			"kill"			},
+[BASE_NR +  29] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [BASE_NR +  30] = { 2,	TF,		SEN(rename),			"rename"		},
 [BASE_NR +  31] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [BASE_NR +  32] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -176,7 +176,7 @@
 [BASE_NR + 153] = { 4,	TS,		SEN(rt_sigaction),		"rt_sigaction"		},
 [BASE_NR + 154] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [BASE_NR + 155] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
-[BASE_NR + 156] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[BASE_NR + 156] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [BASE_NR + 157] = { 0,	TS,		SEN(rt_sigreturn),		"rt_sigreturn"		},
 [BASE_NR + 158] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [BASE_NR + 159] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
@@ -225,13 +225,13 @@
 [BASE_NR + 202] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [BASE_NR + 203] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [BASE_NR + 204] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[BASE_NR + 205] = { 2,	TS,		SEN(kill),			"tkill"			},
+[BASE_NR + 205] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [BASE_NR + 206] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [BASE_NR + 207] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
 [BASE_NR + 208] = { 3,	0,		SEN(sched_getaffinity),		"sched_getaffinity"	},
 [BASE_NR + 209] = { 1,	0,		SEN(set_tid_address),		"set_tid_address"	},
 [BASE_NR + 210] = { 4,	TD,		SEN(fadvise64),			"fadvise64"		},
-[BASE_NR + 211] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[BASE_NR + 211] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [BASE_NR + 212] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [BASE_NR + 213] = { 3,	0,		SEN(lookup_dcookie),		"lookup_dcookie"	},
 [BASE_NR + 214] = { 2,	TM,		SEN(io_setup),			"io_setup"		},
@@ -292,7 +292,7 @@
 [BASE_NR + 269] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [BASE_NR + 270] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [BASE_NR + 271] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[BASE_NR + 272] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[BASE_NR + 272] = { 1,	0,		SEN(unshare),			"unshare"		},
 [BASE_NR + 273] = { 6,	TD,		SEN(splice),			"splice"		},
 [BASE_NR + 274] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [BASE_NR + 275] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
diff --git a/linux/m68k/syscallent.h b/linux/m68k/syscallent.h
index 315dbb6..1876bed 100644
--- a/linux/m68k/syscallent.h
+++ b/linux/m68k/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 5,	TD,		SEN(pread),			"pread64"		},
 [181] = { 5,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -228,7 +228,7 @@
 [218 ... 219] = { },
 [220] = { 3,	TD,		SEN(getdents64),		"getdents64"		},
 [221] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
-[222] = { 2,	TS,		SEN(kill),			"tkill"			},
+[222] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [223] = { 5,	TF,		SEN(setxattr),			"setxattr"		},
 [224] = { 5,	TF,		SEN(setxattr),			"lsetxattr"		},
 [225] = { 5,	TD,		SEN(fsetxattr),			"fsetxattr"		},
@@ -271,7 +271,7 @@
 [262] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [263] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [264] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[265] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[265] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [266] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [267] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [268] = { 6,	TM,		SEN(mbind),			"mbind"			},
@@ -309,7 +309,7 @@
 [300] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [301] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [302] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[303] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[303] = { 1,	0,		SEN(unshare),			"unshare"		},
 [304] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [305] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [306] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/microblaze/syscallent.h b/linux/microblaze/syscallent.h
index c17aad1..5d071bc 100644
--- a/linux/microblaze/syscallent.h
+++ b/linux/microblaze/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 5,	TD,		SEN(pread),			"pread64"		},
 [181] = { 5,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -244,7 +244,7 @@
 [235] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [236] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [237] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[238] = { 2,	TS,		SEN(kill),			"tkill"			},
+[238] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [239] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [240] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [241] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -276,7 +276,7 @@
 [267] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [268] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [269] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[270] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[270] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [271] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [272] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [273] = { 5,	0,		SEN(vserver),			"vserver"		},
@@ -316,7 +316,7 @@
 [307] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [308] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [309] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[310] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[310] = { 1,	0,		SEN(unshare),			"unshare"		},
 [311] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [312] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [313] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/mips/syscallent-compat.h b/linux/mips/syscallent-compat.h
index 918f110..61ccfe4 100644
--- a/linux/mips/syscallent-compat.h
+++ b/linux/mips/syscallent-compat.h
@@ -42,7 +42,7 @@
 [  34] = { 0,	0,	SEN(printargs),		"svr4_nice"	},
 [  35] = { 0,	TF|TSF|TSFA,SEN(printargs),		"svr4_statfs"	},
 [  36] = { 0,	0,	SEN(printargs),		"svr4_sync"	},
-[  37] = { 0,	0,	SEN(printargs),		"svr4_kill"	},
+[  37] = { 0,	TP,	SEN(printargs),		"svr4_kill"	},
 [  38] = { 0,	TD|TFSF|TSFA,SEN(printargs),		"svr4_fstatfs"	},
 [  39] = { 0,	0,	SEN(printargs),		"svr4_setpgrp"	},
 [  40] = { 0,	0,	SEN(printargs),		"svr4_cxenix"	},
@@ -193,7 +193,7 @@
 [1034] = { 0,	0,	SEN(printargs),		"sysv_nice"	},
 [1035] = { 0,	TF|TSF|TSFA,SEN(printargs),		"sysv_statfs"	},
 [1036] = { 0,	0,	SEN(printargs),		"sysv_sync"	},
-[1037] = { 0,	0,	SEN(printargs),		"sysv_kill"	},
+[1037] = { 0,	TP,	SEN(printargs),		"sysv_kill"	},
 [1038] = { 0,	TD|TFSF|TSFA,SEN(printargs),		"sysv_fstatfs"	},
 [1039] = { 0,	0,	SEN(printargs),		"sysv_setpgrp"	},
 [1040] = { 0,	0,	SEN(printargs),		"sysv_syssgi"	},
@@ -378,7 +378,7 @@
 [2034] = { 0,	0,	SEN(printargs),		"bsd43_nice"	},
 [2035] = { 0,	0,	SEN(printargs),		"bsd43_ftime"	},
 [2036] = { 0,	0,	SEN(printargs),		"bsd43_sync"	},
-[2037] = { 0,	0,	SEN(printargs),		"bsd43_kill"	},
+[2037] = { 0,	TP,	SEN(printargs),		"bsd43_kill"	},
 [2038] = { 0,	TF|TST|TSTA,SEN(printargs),		"bsd43_stat"	},
 [2039] = { 0,	0,	SEN(printargs),		"bsd43_oldsetpgrp"	},
 [2040] = { 0,	TF|TLST|TSTA,SEN(printargs),		"bsd43_lstat"	},
@@ -487,7 +487,7 @@
 [2143] = { 0,	0,	SEN(printargs),		"bsd43_sethostid"	},
 [2144] = { 0,	0,	SEN(printargs),		"bsd43_getrlimit"	},
 [2145] = { 0,	0,	SEN(printargs),		"bsd43_setrlimit"	},
-[2146] = { 0,	0,	SEN(printargs),		"bsd43_killpg"	},
+[2146] = { 0,	TP,	SEN(printargs),		"bsd43_killpg"	},
 [2147] = { 0,	0,	SEN(printargs),		"bsd43_shmsys"	},
 [2148] = { 0,	0,	SEN(printargs),		"bsd43_quota"	},
 [2149] = { 0,	0,	SEN(printargs),		"bsd43_qquota"	},
@@ -571,7 +571,7 @@
 [3034] = { 0,	0,	SEN(printargs),		"posix_nice"	},
 [3035] = { 0,	TF|TSF|TSFA,SEN(printargs),		"posix_statfs"	},
 [3036] = { 0,	0,	SEN(printargs),		"posix_sync"	},
-[3037] = { 0,	0,	SEN(printargs),		"posix_kill"	},
+[3037] = { 0,	TP,	SEN(printargs),		"posix_kill"	},
 [3038] = { 0,	TD|TFSF|TSFA,SEN(printargs),		"posix_fstatfs"	},
 [3039] = { 0,	0,	SEN(printargs),		"posix_getpgrp"	},
 [3040] = { 0,	0,	SEN(printargs),		"posix_syssgi"	},
diff --git a/linux/mips/syscallent-n32.h b/linux/mips/syscallent-n32.h
index 38773f8..cfd199d 100644
--- a/linux/mips/syscallent-n32.h
+++ b/linux/mips/syscallent-n32.h
@@ -68,7 +68,7 @@
 [BASE_NR +  57] = { 3,	TF|TP|TSD|SE|SI,	SEN(execve),			"execve"		},
 [BASE_NR +  58] = { 1,	TP|SE,		SEN(exit),			"exit"			},
 [BASE_NR +  59] = { 4,	TP,		SEN(wait4),			"wait4"			},
-[BASE_NR +  60] = { 2,	TS,		SEN(kill),			"kill"			},
+[BASE_NR +  60] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [BASE_NR +  61] = { 1,	0,		SEN(uname),			"uname"			},
 [BASE_NR +  62] = { 3,	TI,		SEN(semget),			"semget"		},
 [BASE_NR +  63] = { 3,	TI,		SEN(semop),			"semop"			},
@@ -135,7 +135,7 @@
 [BASE_NR + 124] = { 2,	TC,		SEN(capset),			"capset"		},
 [BASE_NR + 125] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [BASE_NR + 126] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[BASE_NR + 127] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[BASE_NR + 127] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [BASE_NR + 128] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [BASE_NR + 129] = { 2,	TS,		SEN(sigaltstack),		"sigaltstack"		},
 [BASE_NR + 130] = { 2,	TF,		SEN(utime),			"utime"			},
@@ -200,7 +200,7 @@
 [BASE_NR + 189] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [BASE_NR + 190] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [BASE_NR + 191] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[BASE_NR + 192] = { 2,	TS,		SEN(kill),			"tkill"			},
+[BASE_NR + 192] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [BASE_NR + 193] = { 1,	TCL,		SEN(time),			"time"			},
 [BASE_NR + 194] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [BASE_NR + 195] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -237,7 +237,7 @@
 [BASE_NR + 226] = { 2,	TCL,		SEN(clock_gettime32),		"clock_gettime"		},
 [BASE_NR + 227] = { 2,	TCL,		SEN(clock_getres_time32),	"clock_getres"		},
 [BASE_NR + 228] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
-[BASE_NR + 229] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[BASE_NR + 229] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [BASE_NR + 230] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [BASE_NR + 231] = { 6,	TM,		SEN(mbind),			"mbind"			},
 [BASE_NR + 232] = { 5,	TM,		SEN(get_mempolicy),		"get_mempolicy"		},
@@ -274,7 +274,7 @@
 [BASE_NR + 263] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [BASE_NR + 264] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [BASE_NR + 265] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[BASE_NR + 266] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[BASE_NR + 266] = { 1,	0,		SEN(unshare),			"unshare"		},
 [BASE_NR + 267] = { 6,	TD,		SEN(splice),			"splice"		},
 [BASE_NR + 268] = { 4,	TD,		SEN(sync_file_range),		"sync_file_range"	},
 [BASE_NR + 269] = { 4,	TD,		SEN(tee),			"tee"			},
diff --git a/linux/mips/syscallent-n64.h b/linux/mips/syscallent-n64.h
index 0dfd64e..7ef6700 100644
--- a/linux/mips/syscallent-n64.h
+++ b/linux/mips/syscallent-n64.h
@@ -68,7 +68,7 @@
 [BASE_NR +  57] = { 3,	TF|TP|TSD|SE|SI,	SEN(execve),			"execve"		},
 [BASE_NR +  58] = { 1,	TP|SE,		SEN(exit),			"exit"			},
 [BASE_NR +  59] = { 4,	TP,		SEN(wait4),			"wait4"			},
-[BASE_NR +  60] = { 2,	TS,		SEN(kill),			"kill"			},
+[BASE_NR +  60] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [BASE_NR +  61] = { 1,	0,		SEN(uname),			"uname"			},
 [BASE_NR +  62] = { 3,	TI,		SEN(semget),			"semget"		},
 [BASE_NR +  63] = { 3,	TI,		SEN(semop),			"semop"			},
@@ -135,7 +135,7 @@
 [BASE_NR + 124] = { 2,	TC,		SEN(capset),			"capset"		},
 [BASE_NR + 125] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [BASE_NR + 126] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[BASE_NR + 127] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[BASE_NR + 127] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [BASE_NR + 128] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [BASE_NR + 129] = { 2,	TS,		SEN(sigaltstack),		"sigaltstack"		},
 [BASE_NR + 130] = { 2,	TF,		SEN(utime),			"utime"			},
@@ -200,7 +200,7 @@
 [BASE_NR + 189] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [BASE_NR + 190] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [BASE_NR + 191] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[BASE_NR + 192] = { 2,	TS,		SEN(kill),			"tkill"			},
+[BASE_NR + 192] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [BASE_NR + 193] = { 1,	TCL,		SEN(time),			"time"			},
 [BASE_NR + 194] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [BASE_NR + 195] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -233,7 +233,7 @@
 [BASE_NR + 222] = { 2,	TCL,		SEN(clock_gettime64),		"clock_gettime"		},
 [BASE_NR + 223] = { 2,	TCL,		SEN(clock_getres_time64),	"clock_getres"		},
 [BASE_NR + 224] = { 4,	0,		SEN(clock_nanosleep_time64),	"clock_nanosleep"	},
-[BASE_NR + 225] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[BASE_NR + 225] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [BASE_NR + 226] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [BASE_NR + 227] = { 6,	TM,		SEN(mbind),			"mbind"			},
 [BASE_NR + 228] = { 5,	TM,		SEN(get_mempolicy),		"get_mempolicy"		},
@@ -270,7 +270,7 @@
 [BASE_NR + 259] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [BASE_NR + 260] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [BASE_NR + 261] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[BASE_NR + 262] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[BASE_NR + 262] = { 1,	0,		SEN(unshare),			"unshare"		},
 [BASE_NR + 263] = { 6,	TD,		SEN(splice),			"splice"		},
 [BASE_NR + 264] = { 4,	TD,		SEN(sync_file_range),		"sync_file_range"	},
 [BASE_NR + 265] = { 4,	TD,		SEN(tee),			"tee"			},
diff --git a/linux/mips/syscallent-o32.h b/linux/mips/syscallent-o32.h
index f4e67e8..d2b26f7 100644
--- a/linux/mips/syscallent-o32.h
+++ b/linux/mips/syscallent-o32.h
@@ -46,7 +46,7 @@
 [BASE_NR +  34] = { 1,	0,		SEN(nice),			"nice"			},
 [BASE_NR +  35] = { 1,	0,		SEN(ftime),			"ftime"			},
 [BASE_NR +  36] = { 0,	0,		SEN(sync),			"sync"			},
-[BASE_NR +  37] = { 2,	TS,		SEN(kill),			"kill"			},
+[BASE_NR +  37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [BASE_NR +  38] = { 2,	TF,		SEN(rename),			"rename"		},
 [BASE_NR +  39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [BASE_NR +  40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -207,7 +207,7 @@
 [BASE_NR + 195] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [BASE_NR + 196] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [BASE_NR + 197] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[BASE_NR + 198] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[BASE_NR + 198] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [BASE_NR + 199] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [BASE_NR + 200] = { 6,	TD,		SEN(pread),			"pread64"		},
 [BASE_NR + 201] = { 6,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -245,7 +245,7 @@
 [BASE_NR + 233] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [BASE_NR + 234] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [BASE_NR + 235] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[BASE_NR + 236] = { 2,	TS,		SEN(kill),			"tkill"			},
+[BASE_NR + 236] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [BASE_NR + 237] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [BASE_NR + 238] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [BASE_NR + 239] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -275,7 +275,7 @@
 [BASE_NR + 263] = { 2,	TCL,		SEN(clock_gettime32),		"clock_gettime"		},
 [BASE_NR + 264] = { 2,	TCL,		SEN(clock_getres_time32),	"clock_getres"		},
 [BASE_NR + 265] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
-[BASE_NR + 266] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[BASE_NR + 266] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [BASE_NR + 267] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [BASE_NR + 268] = { 6,	TM,		SEN(mbind),			"mbind"			},
 [BASE_NR + 269] = { 5,	TM,		SEN(get_mempolicy),		"get_mempolicy"		},
@@ -312,7 +312,7 @@
 [BASE_NR + 300] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [BASE_NR + 301] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [BASE_NR + 302] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[BASE_NR + 303] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[BASE_NR + 303] = { 1,	0,		SEN(unshare),			"unshare"		},
 [BASE_NR + 304] = { 6,	TD,		SEN(splice),			"splice"		},
 [BASE_NR + 305] = { 7,	TD,		SEN(sync_file_range),		"sync_file_range"	},
 [BASE_NR + 306] = { 4,	TD,		SEN(tee),			"tee"			},
diff --git a/linux/powerpc/syscallent.h b/linux/powerpc/syscallent.h
index 6958eda..b0962b4 100644
--- a/linux/powerpc/syscallent.h
+++ b/linux/powerpc/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -184,7 +184,7 @@
 [174] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [175] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [176] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[177] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[177] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [178] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [179] = { 6,	TD,		SEN(pread),			"pread64"		},
 [180] = { 6,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -215,7 +215,7 @@
 [205] = { 3,	TM,		SEN(madvise),			"madvise"		},
 [206] = { 3,	TM,		SEN(mincore),			"mincore"		},
 [207] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
-[208] = { 2,	TS,		SEN(kill),			"tkill"			},
+[208] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [209] = { 5,	TF,		SEN(setxattr),			"setxattr"		},
 [210] = { 5,	TF,		SEN(setxattr),			"lsetxattr"		},
 [211] = { 5,	TD,		SEN(fsetxattr),			"fsetxattr"		},
@@ -257,7 +257,7 @@
 [247] = { 2,	TCL,		SEN(clock_getres_time32),	"clock_getres"		},
 [248] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [249] = { 2,	0,		SEN(printargs),			"swapcontext"		},
-[250] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[250] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [251] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [252] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [253] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
@@ -289,7 +289,7 @@
 [279] = { 4,	0,		SEN(printargs),			"spu_create"		},
 [280] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [281] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[282] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[282] = { 1,	0,		SEN(unshare),			"unshare"		},
 [283] = { 6,	TD,		SEN(splice),			"splice"		},
 [284] = { 4,	TD,		SEN(tee),			"tee"			},
 [285] = { 4,	TD,		SEN(vmsplice),			"vmsplice"		},
diff --git a/linux/powerpc64/syscallent.h b/linux/powerpc64/syscallent.h
index 608caed..1a0dfb5 100644
--- a/linux/powerpc64/syscallent.h
+++ b/linux/powerpc64/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -184,7 +184,7 @@
 [174] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [175] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [176] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[177] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[177] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [178] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [179] = { 4,	TD,		SEN(pread),			"pread64"		},
 [180] = { 4,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -210,7 +210,7 @@
 [205] = { 3,	TM,		SEN(madvise),			"madvise"		},
 [206] = { 3,	TM,		SEN(mincore),			"mincore"		},
 [207] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
-[208] = { 2,	TS,		SEN(kill),			"tkill"			},
+[208] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [209] = { 5,	TF,		SEN(setxattr),			"setxattr"		},
 [210] = { 5,	TF,		SEN(setxattr),			"lsetxattr"		},
 [211] = { 5,	TD,		SEN(fsetxattr),			"fsetxattr"		},
@@ -252,7 +252,7 @@
 [247] = { 2,	TCL,		SEN(clock_getres_time64),	"clock_getres"		},
 [248] = { 4,	0,		SEN(clock_nanosleep_time64),	"clock_nanosleep"	},
 [249] = { 2,	0,		SEN(printargs),			"swapcontext"		},
-[250] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[250] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [251] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [252] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [253] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
@@ -284,7 +284,7 @@
 [279] = { 4,	0,		SEN(printargs),			"spu_create"		},
 [280] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [281] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[282] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[282] = { 1,	0,		SEN(unshare),			"unshare"		},
 [283] = { 6,	TD,		SEN(splice),			"splice"		},
 [284] = { 4,	TD,		SEN(tee),			"tee"			},
 [285] = { 4,	TD,		SEN(vmsplice),			"vmsplice"		},
diff --git a/linux/s390/syscallent.h b/linux/s390/syscallent.h
index c61ad77..105089f 100644
--- a/linux/s390/syscallent.h
+++ b/linux/s390/syscallent.h
@@ -46,7 +46,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { },
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -187,7 +187,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 5,	TD,		SEN(pread),			"pread64"		},
 [181] = { 5,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -246,11 +246,11 @@
 [234] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [235] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
 [236] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
-[237] = { 2,	TS,		SEN(kill),			"tkill"			},
+[237] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [238] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [239] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
 [240] = { 3,	0,		SEN(sched_getaffinity),		"sched_getaffinity"	},
-[241] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[241] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [242] = { },
 [243] = { 2,	TM,		SEN(io_setup),			"io_setup"		},
 [244] = { 1,	TM,		SEN(io_destroy),		"io_destroy"		},
@@ -312,7 +312,7 @@
 [300] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [301] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [302] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[303] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[303] = { 1,	0,		SEN(unshare),			"unshare"		},
 [304] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [305] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [306] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/s390x/syscallent.h b/linux/s390x/syscallent.h
index c493757..e9cf57a 100644
--- a/linux/s390x/syscallent.h
+++ b/linux/s390x/syscallent.h
@@ -45,7 +45,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { },
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -176,7 +176,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 4,	TD,		SEN(pread),			"pread64"		},
 [181] = { 4,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -230,11 +230,11 @@
 [234] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [235] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
 [236] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
-[237] = { 2,	TS,		SEN(kill),			"tkill"			},
+[237] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [238] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [239] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
 [240] = { 3,	0,		SEN(sched_getaffinity),		"sched_getaffinity"	},
-[241] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[241] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [242] = { },
 [243] = { 2,	TM,		SEN(io_setup),			"io_setup"		},
 [244] = { 1,	TM,		SEN(io_destroy),		"io_destroy"		},
@@ -296,7 +296,7 @@
 [300] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [301] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [302] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[303] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[303] = { 1,	0,		SEN(unshare),			"unshare"		},
 [304] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [305] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [306] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/sh/syscallent.h b/linux/sh/syscallent.h
index 61eb728..70dc7da 100644
--- a/linux/sh/syscallent.h
+++ b/linux/sh/syscallent.h
@@ -46,7 +46,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -187,7 +187,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 6,	TD,		SEN(pread),			"pread64"		},
 [181] = { 6,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -245,7 +245,7 @@
 [235] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [236] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [237] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[238] = { 2,	TS,		SEN(kill),			"tkill"			},
+[238] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [239] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [240] = { 6,	0,		SEN(futex_time32),		"futex"			},
 [241] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -276,7 +276,7 @@
 [267] = { 4,	0,		SEN(clock_nanosleep_time32),	"clock_nanosleep"	},
 [268] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [269] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[270] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[270] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [271] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [272] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [273] = { },
@@ -316,7 +316,7 @@
 [307] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [308] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [309] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[310] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[310] = { 1,	0,		SEN(unshare),			"unshare"		},
 [311] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [312] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [313] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/sh64/syscallent.h b/linux/sh64/syscallent.h
index f681635..eff5dc0 100644
--- a/linux/sh64/syscallent.h
+++ b/linux/sh64/syscallent.h
@@ -44,7 +44,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 0,	0,		SEN(ftime),			"ftime"			},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF,		SEN(rename),			"rename"		},
 [ 39] = { 2,	TF,		SEN(mkdir),			"mkdir"			},
 [ 40] = { 1,	TF,		SEN(rmdir),			"rmdir"			},
@@ -185,7 +185,7 @@
 [175] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [176] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [177] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[178] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[178] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [179] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [180] = { 4,	TD,		SEN(pread),			"pread64"		},
 [181] = { 4,	TD,		SEN(pwrite),			"pwrite64"		},
@@ -271,7 +271,7 @@
 [263] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [264] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [265] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[266] = { 2,	TS,		SEN(kill),			"tkill"			},
+[266] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [267] = { 4,	TD|TN,		SEN(sendfile64),		"sendfile64"		},
 [268] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [269] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -302,7 +302,7 @@
 [295] = { 4,	0,		SEN(clock_nanosleep_time64),	"clock_nanosleep"	},
 [296] = { 3,	TF|TSF|TSFA,	SEN(statfs64),			"statfs64"		},
 [297] = { 3,	TD|TFSF|TSFA,	SEN(fstatfs64),			"fstatfs64"		},
-[298] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[298] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [299] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [300] = { 4,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
 [301] = { },
@@ -342,7 +342,7 @@
 [335] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [336] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [337] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[338] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[338] = { 1,	0,		SEN(unshare),			"unshare"		},
 [339] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [340] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [341] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/sparc/syscallent.h b/linux/sparc/syscallent.h
index 3417e7f..8c8bd18 100644
--- a/linux/sparc/syscallent.h
+++ b/linux/sparc/syscallent.h
@@ -42,7 +42,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { 3,	TF,		SEN(chown),			"chown32"		},
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF|TST|TSTA,	SEN(stat),			"stat"			},
 [ 39] = { 4,	TD|TN,		SEN(sendfile),			"sendfile"		},
 [ 40] = { 2,	TF|TLST|TSTA,	SEN(lstat),			"lstat"			},
@@ -111,7 +111,7 @@
 [103] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [104] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [105] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[106] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[106] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [107] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [108] = { 3,	TC,		SEN(setresuid),			"setresuid32"		},
 [109] = { 3,	TC,		SEN(getresuid),			"getresuid32"		},
@@ -192,7 +192,7 @@
 [184] = { 5,	0,		SEN(query_module),		"query_module"		},
 [185] = { 2,	0,		SEN(setpgid),			"setpgid"		},
 [186] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[187] = { 2,	TS,		SEN(kill),			"tkill"			},
+[187] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [188] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [189] = { 1,	0,		SEN(uname),			"uname"			},
 [190] = { 3,	0,		SEN(init_module),		"init_module"		},
@@ -216,7 +216,7 @@
 [208] = { 4,	0,		SEN(lookup_dcookie),		"lookup_dcookie"	},
 [209] = { 5,	TD,		SEN(fadvise64),			"fadvise64"		},
 [210] = { 6,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
-[211] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[211] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [212] = { 3,	TP,		SEN(waitpid),			"waitpid"		},
 [213] = { 1,	TF,		SEN(swapoff),			"swapoff"		},
 [214] = { 1,	0,		SEN(sysinfo),			"sysinfo"		},
@@ -304,7 +304,7 @@
 [296] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [297] = { 6,	TD,		SEN(pselect6_time32),		"pselect6"		},
 [298] = { 5,	TD,		SEN(ppoll_time32),		"ppoll"			},
-[299] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[299] = { 1,	0,		SEN(unshare),			"unshare"		},
 [300] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [301] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [302] = { 4,	TM,		SEN(migrate_pages),		"migrate_pages"		},
diff --git a/linux/sparc64/syscallent.h b/linux/sparc64/syscallent.h
index dd77685..0e0e0c4 100644
--- a/linux/sparc64/syscallent.h
+++ b/linux/sparc64/syscallent.h
@@ -41,7 +41,7 @@
 [ 34] = { 1,	0,		SEN(nice),			"nice"			},
 [ 35] = { },
 [ 36] = { 0,	0,		SEN(sync),			"sync"			},
-[ 37] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 37] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 38] = { 2,	TF|TST|TSTA,	SEN(stat),			"stat"			},
 [ 39] = { 4,	TD|TN,		SEN(sendfile),			"sendfile"		},
 [ 40] = { 2,	TF|TLST|TSTA,	SEN(lstat),			"lstat"			},
@@ -109,7 +109,7 @@
 [103] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [104] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [105] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[106] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[106] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [107] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [108] = { 3,	TC,		SEN(setresuid),			"setresuid"		},
 [109] = { 3,	TC,		SEN(getresuid),			"getresuid"		},
@@ -190,7 +190,7 @@
 [184] = { 5,	0,		SEN(query_module),		"query_module"		},
 [185] = { 2,	0,		SEN(setpgid),			"setpgid"		},
 [186] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[187] = { 2,	TS,		SEN(kill),			"tkill"			},
+[187] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [188] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [189] = { 1,	0,		SEN(uname),			"uname"			},
 [190] = { 3,	0,		SEN(init_module),		"init_module"		},
@@ -214,7 +214,7 @@
 [208] = { 3,	0,		SEN(lookup_dcookie),		"lookup_dcookie"	},
 [209] = { 4,	TD,		SEN(fadvise64),			"fadvise64"		},
 [210] = { 4,	TD,		SEN(fadvise64_64),		"fadvise64_64"		},
-[211] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[211] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [212] = { 3,	TP,		SEN(waitpid),			"waitpid"		},
 [213] = { 1,	TF,		SEN(swapoff),			"swapoff"		},
 [214] = { 1,	0,		SEN(sysinfo),			"sysinfo"		},
@@ -302,7 +302,7 @@
 [296] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [297] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [298] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[299] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[299] = { 1,	0,		SEN(unshare),			"unshare"		},
 [300] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [301] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [302] = { 4,	TM,		SEN(migrate_pages),		"migrate_pages"		},
diff --git a/linux/syscallent-common.h b/linux/syscallent-common.h
index 0d3a4db..0ff671e 100644
--- a/linux/syscallent-common.h
+++ b/linux/syscallent-common.h
@@ -8,7 +8,7 @@
 #ifndef BASE_NR
 # define BASE_NR 0
 #endif
-[BASE_NR + 424] = { 4,	TD|TS,		SEN(pidfd_send_signal),		"pidfd_send_signal"	},
+[BASE_NR + 424] = { 4,	TD|TS|TP,		SEN(pidfd_send_signal),		"pidfd_send_signal"	},
 [BASE_NR + 425] = { 2,	TD,		SEN(io_uring_setup),		"io_uring_setup"	},
 [BASE_NR + 426] = { 6,	TD|TS,		SEN(io_uring_enter),		"io_uring_enter"	},
 [BASE_NR + 427] = { 4,	TD|TM,		SEN(io_uring_register),		"io_uring_register"	},
diff --git a/linux/x32/syscallent.h b/linux/x32/syscallent.h
index 30e295f..db8ecfd 100644
--- a/linux/x32/syscallent.h
+++ b/linux/x32/syscallent.h
@@ -67,7 +67,7 @@
 [ 59] = { 3,	TF|TP|SE|SI,	SEN(printargs),			"execve#64"		},
 [ 60] = { 1,	TP|SE,		SEN(exit),			"exit"			},
 [ 61] = { 4,	TP,		SEN(wait4),			"wait4"			},
-[ 62] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 62] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 63] = { 1,	0,		SEN(uname),			"uname"			},
 [ 64] = { 3,	TI,		SEN(semget),			"semget"		},
 [ 65] = { 3,	TI,		SEN(semop),			"semop"			},
@@ -134,7 +134,7 @@
 [126] = { 2,	TC,		SEN(capset),			"capset"		},
 [127] = { 2,	TS,		SEN(printargs),			"rt_sigpending#64"	},
 [128] = { 4,	TS,		SEN(printargs),			"rt_sigtimedwait#64"	},
-[129] = { 3,	TS,		SEN(printargs),			"rt_sigqueueinfo#64"	},
+[129] = { 3,	TS|TP,		SEN(printargs),			"rt_sigqueueinfo#64"	},
 [130] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [131] = { 2,	TS,		SEN(printargs),			"sigaltstack#64"	},
 [132] = { 2,	TF,		SEN(utime),			"utime"			},
@@ -163,7 +163,7 @@
 [155] = { 2,	TF,		SEN(pivotroot),			"pivot_root"		},
 [156] = { 1,	0,		SEN(printargs),			"_sysctl#64"		},
 [157] = { 5,	TC,		SEN(prctl),			"prctl"			},
-[158] = { 2,	TP,		SEN(arch_prctl),		"arch_prctl"		},
+[158] = { 2,	0,		SEN(arch_prctl),		"arch_prctl"		},
 [159] = { 1,	TCL,		SEN(adjtimex64),		"adjtimex"		},
 [160] = { 2,	0,		SEN(setrlimit),			"setrlimit"		},
 [161] = { 1,	TF,		SEN(chroot),			"chroot"		},
@@ -205,7 +205,7 @@
 [197] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [198] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [199] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[200] = { 2,	TS,		SEN(kill),			"tkill"			},
+[200] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [201] = { 1,	TCL,		SEN(time),			"time"			},
 [202] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [203] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -239,7 +239,7 @@
 [231] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [232] = { 4,	TD,		SEN(epoll_wait),		"epoll_wait"		},
 [233] = { 4,	TD,		SEN(epoll_ctl),			"epoll_ctl"		},
-[234] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[234] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [235] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [236] = { 5,	0,		SEN(printargs),			"vserver#64"		},
 [237] = { 6,	TM,		SEN(mbind),			"mbind"			},
@@ -277,7 +277,7 @@
 [269] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [270] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [271] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[272] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[272] = { 1,	0,		SEN(unshare),			"unshare"		},
 [273] = { 2,	0,		SEN(printargs),			"set_robust_list#64"	},
 [274] = { 3,	0,		SEN(printargs),			"get_robust_list#64"	},
 [275] = { 6,	TD,		SEN(splice),			"splice"		},
@@ -358,7 +358,7 @@
 [521] = { 4,	CST,		SEN(ptrace),			"ptrace"		},
 [522] = { 2,	CST|TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [523] = { 4,	CST|TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[524] = { 3,	CST|TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[524] = { 3,	CST|TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [525] = { 2,	CST|TS,		SEN(sigaltstack),		"sigaltstack"		},
 [526] = { 3,	CST,		SEN(timer_create),		"timer_create"		},
 [527] = { 2,	CST|TD,		SEN(mq_notify),			"mq_notify"		},
diff --git a/linux/x86_64/syscallent.h b/linux/x86_64/syscallent.h
index 8423c7d..c69a5aa 100644
--- a/linux/x86_64/syscallent.h
+++ b/linux/x86_64/syscallent.h
@@ -67,7 +67,7 @@
 [ 59] = { 3,	TF|TP|TSD|SE|SI,	SEN(execve),			"execve"		},
 [ 60] = { 1,	TP|SE,		SEN(exit),			"exit"			},
 [ 61] = { 4,	TP,		SEN(wait4),			"wait4"			},
-[ 62] = { 2,	TS,		SEN(kill),			"kill"			},
+[ 62] = { 2,	TS|TP,		SEN(kill),			"kill"			},
 [ 63] = { 1,	0,		SEN(uname),			"uname"			},
 [ 64] = { 3,	TI,		SEN(semget),			"semget"		},
 [ 65] = { 3,	TI,		SEN(semop),			"semop"			},
@@ -134,7 +134,7 @@
 [126] = { 2,	TC,		SEN(capset),			"capset"		},
 [127] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [128] = { 4,	TS,		SEN(rt_sigtimedwait_time64),	"rt_sigtimedwait"	},
-[129] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[129] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [130] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [131] = { 2,	TS,		SEN(sigaltstack),		"sigaltstack"		},
 [132] = { 2,	TF,		SEN(utime),			"utime"			},
@@ -163,7 +163,7 @@
 [155] = { 2,	TF,		SEN(pivotroot),			"pivot_root"		},
 [156] = { 1,	0,		SEN(sysctl),			"_sysctl"		},
 [157] = { 5,	TC,		SEN(prctl),			"prctl"			},
-[158] = { 2,	TP,		SEN(arch_prctl),		"arch_prctl"		},
+[158] = { 2,	0,		SEN(arch_prctl),		"arch_prctl"		},
 [159] = { 1,	TCL,		SEN(adjtimex64),		"adjtimex"		},
 [160] = { 2,	0,		SEN(setrlimit),			"setrlimit"		},
 [161] = { 1,	TF,		SEN(chroot),			"chroot"		},
@@ -205,7 +205,7 @@
 [197] = { 2,	TF,		SEN(removexattr),		"removexattr"		},
 [198] = { 2,	TF,		SEN(removexattr),		"lremovexattr"		},
 [199] = { 2,	TD,		SEN(fremovexattr),		"fremovexattr"		},
-[200] = { 2,	TS,		SEN(kill),			"tkill"			},
+[200] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
 [201] = { 1,	TCL,		SEN(time),			"time"			},
 [202] = { 6,	0,		SEN(futex_time64),		"futex"			},
 [203] = { 3,	0,		SEN(sched_setaffinity),		"sched_setaffinity"	},
@@ -239,7 +239,7 @@
 [231] = { 1,	TP|SE,		SEN(exit),			"exit_group"		},
 [232] = { 4,	TD,		SEN(epoll_wait),		"epoll_wait"		},
 [233] = { 4,	TD,		SEN(epoll_ctl),			"epoll_ctl"		},
-[234] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[234] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [235] = { 2,	TF,		SEN(utimes),			"utimes"		},
 [236] = { 5,	0,		SEN(vserver),			"vserver"		},
 [237] = { 6,	TM,		SEN(mbind),			"mbind"			},
@@ -277,7 +277,7 @@
 [269] = { 3,	TD|TF,		SEN(faccessat),			"faccessat"		},
 [270] = { 6,	TD,		SEN(pselect6_time64),		"pselect6"		},
 [271] = { 5,	TD,		SEN(ppoll_time64),		"ppoll"			},
-[272] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[272] = { 1,	0,		SEN(unshare),			"unshare"		},
 [273] = { 2,	0,		SEN(set_robust_list),		"set_robust_list"	},
 [274] = { 3,	0,		SEN(get_robust_list),		"get_robust_list"	},
 [275] = { 6,	TD,		SEN(splice),			"splice"		},
diff --git a/linux/xtensa/syscallent.h b/linux/xtensa/syscallent.h
index 385630a..61f1dd4 100644
--- a/linux/xtensa/syscallent.h
+++ b/linux/xtensa/syscallent.h
@@ -123,9 +123,9 @@
 [120] = { 0,	PU|NF,		SEN(getpid),			"getpid"		},
 [121] = { 4,	TP,		SEN(wait4),			"wait4"			},
 [122] = { 5,	TP,		SEN(waitid),			"waitid"		},
-[123] = { 2,	TS,		SEN(kill),			"kill"			},
-[124] = { 2,	TS,		SEN(kill),			"tkill"			},
-[125] = { 3,	TS,		SEN(tgkill),			"tgkill"		},
+[123] = { 2,	TS|TP,		SEN(kill),			"kill"			},
+[124] = { 2,	TS|TP,		SEN(kill),			"tkill"			},
+[125] = { 3,	TS|TP,		SEN(tgkill),			"tgkill"		},
 [126] = { 1,	0,		SEN(set_tid_address),		"set_tid_address"	},
 [127] = { 0,	PU|NF,		SEN(gettid),			"gettid"		},
 [128] = { 0,	0,		SEN(setsid),			"setsid"		},
@@ -227,7 +227,7 @@
 [227] = { 4,	TS,		SEN(rt_sigprocmask),		"rt_sigprocmask"	},
 [228] = { 2,	TS,		SEN(rt_sigpending),		"rt_sigpending"		},
 [229] = { 4,	TS,		SEN(rt_sigtimedwait_time32),	"rt_sigtimedwait"	},
-[230] = { 3,	TS,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
+[230] = { 3,	TS|TP,		SEN(rt_sigqueueinfo),		"rt_sigqueueinfo"	},
 [231] = { 2,	TS,		SEN(rt_sigsuspend),		"rt_sigsuspend"		},
 [232] = { 4,	TD,		SEN(mq_open),			"mq_open"		},
 [233] = { 1,	0,		SEN(mq_unlink),			"mq_unlink"		},
@@ -263,7 +263,7 @@
 [263] = { 6,	TM,		SEN(mbind),			"mbind"			},
 [264] = { 5,	TM,		SEN(get_mempolicy),		"get_mempolicy"		},
 [265] = { 3,	TM,		SEN(set_mempolicy),		"set_mempolicy"		},
-[266] = { 1,	TP,		SEN(unshare),			"unshare"		},
+[266] = { 1,	0,		SEN(unshare),			"unshare"		},
 [267] = { 6,	TM,		SEN(move_pages),		"move_pages"		},
 [268] = { 6,	TD,		SEN(splice),			"splice"		},
 [269] = { 4,	TD,		SEN(tee),			"tee"			},
diff --git a/strace.1.in b/strace.1.in
index 7564fc2..fed8b5a 100644
--- a/strace.1.in
+++ b/strace.1.in
@@ -518,8 +518,8 @@ is deprecated.
 .B %process
 .TQ
 .B process
-Trace all system calls which involve process management.  This
-is useful for watching the fork, wait, and exec steps of a process.
+Trace system calls associated with process lifecycle
+(creation, exec, termination).
 The syntax without a preceding percent sign
 .RB (\[dq] "-e trace" = process \[dq])
 is deprecated.
-- 
2.1.4

